api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/DiagnosticReportHandler.java

```diff
@@ -1,12 +1,19 @@
 package org.openmrs.module.fhir.api.diagnosticreport;
 
 import ca.uhn.fhir.model.dstu2.resource.DiagnosticReport;
+import org.openmrs.Obs;
+
+import java.util.List;
+import java.util.Set;
 
 public interface DiagnosticReportHandler {
-	
-	public DiagnosticReport generateFHIRDiagnosticReport(DiagnosticReportTemplate omrsDiagnosticReport);
-	
-	public DiagnosticReportTemplate generateOpenMRSDiagnosticReport(DiagnosticReport fhirDiagnosticReport);
-	
+
 	public String getId();
+
+	public Set<Obs> saveObs(List<Obs> result);
```

Comment:
@harsha89 We want to let the user implement ( or handler) how should the Obs to be save. That will different form Handler to Handler. As you can see, it'll pass a list of OpenMRS Obs. Then user can save them and return saved Obs back to FHIRDiagReportUtil class. Then it will include that in OpenMRS Encounter and return the Encounter back to the Service. Service will save it their.